random-distribution-helpers.go

```diff
@@ -14,60 +14,66 @@ type RandomDistribution interface {
 	String() string
 }
 
-var _ RandomDistribution = RandomConstDistribution{}
+var _ RandomDistribution = &RandomConstDistribution{}
 
 type RandomConstDistribution struct {
 	value time.Duration
 }
 
-func (constDist RandomConstDistribution) Sample() time.Duration {
+func (constDist *RandomConstDistribution) Sample() time.Duration {
 	return constDist.value
 }
 
-func (constDist RandomConstDistribution) String() string {
-	return fmt.Sprintf("Constant value: %d.", constDist.value)
+func (constDist *RandomConstDistribution) String() string {
+	return fmt.Sprintf("Constant value: %v ms.", int64(constDist.value/time.Millisecond))
 }
 
-var _ RandomDistribution = RandomEqualDistribution{}
+var _ RandomDistribution = &RandomEqualDistribution{}
 
 type RandomEqualDistribution struct {
 	min time.Duration
 	max time.Duration
 }
 
-func (equalDist RandomEqualDistribution) Sample() time.Duration {
-	return time.Duration(rand.Int63n(int64(equalDist.max) - int64(equalDist.min)) + int64(equalDist.min))
+func (equalDist *RandomEqualDistribution) Sample() time.Duration {
+	return time.Duration(rand.Int63n(int64(equalDist.max)-int64(equalDist.min)) + int64(equalDist.min))
 }
 
-func (equalDist RandomEqualDistribution) String() string {
-	return fmt.Sprintf("Equal distribution between %d and %d.", equalDist.min, equalDist.max)
+func (equalDist *RandomEqualDistribution) String() string {
+	return fmt.Sprintf("Equal distribution between %v ms and %v ms.",
+		int64(equalDist.min/time.Millisecond), int64(equalDist.max/time.Millisecond))
```

Comment:
See above comment about printing time.Duration values.

Reply:
Also check the other places where you do that.

Reply:
See comment above

Reply:
See reply above